src/Compilers/CSharp/Test/Semantic/Semantics/TargetTypedObjectCreationTests.cs

```diff
@@ -0,0 +1,3926 @@
+                // (6,17): error CS8130: Cannot infer the type of implicitly-typed deconstruction variable '_'.
+                //         var (_, _) = new();
+                Diagnostic(ErrorCode.ERR_TypeInferenceFailedForImplicitlyTypedDeconstructionVariable, "_").WithArguments("_").WithLocation(6, 17),
+                // (6,17): error CS8183: Cannot infer the type of implicitly-typed discard.
+                //         var (_, _) = new();
+                Diagnostic(ErrorCode.ERR_DiscardTypeInferenceFailed, "_").WithLocation(6, 17),
+                // (7,26): error CS8755: There is no target type for new() expression
+                //         (var _, var _) = new();
+                Diagnostic(ErrorCode.ERR_TypelessNewNoTargetType, "new()").WithLocation(7, 26),
+                // (7,26): error CS8131: Deconstruct assignment requires an expression with a type on the right-hand-side.
+                //         (var _, var _) = new();
+                Diagnostic(ErrorCode.ERR_DeconstructRequiresExpression, "new()").WithLocation(7, 26),
+                // (7,10): error CS8130: Cannot infer the type of implicitly-typed deconstruction variable '_'.
+                //         (var _, var _) = new();
+                Diagnostic(ErrorCode.ERR_TypeInferenceFailedForImplicitlyTypedDeconstructionVariable, "var _").WithArguments("_").WithLocation(7, 10),
+                // (7,10): error CS8183: Cannot infer the type of implicitly-typed discard.
+                //         (var _, var _) = new();
+                Diagnostic(ErrorCode.ERR_DiscardTypeInferenceFailed, "var _").WithLocation(7, 10),
+                // (7,17): error CS8130: Cannot infer the type of implicitly-typed deconstruction variable '_'.
+                //         (var _, var _) = new();
+                Diagnostic(ErrorCode.ERR_TypeInferenceFailedForImplicitlyTypedDeconstructionVariable, "var _").WithArguments("_").WithLocation(7, 17),
+                // (7,17): error CS8183: Cannot infer the type of implicitly-typed discard.
+                //         (var _, var _) = new();
+                Diagnostic(ErrorCode.ERR_DiscardTypeInferenceFailed, "var _").WithLocation(7, 17),
+                // (8,22): error CS8755: There is no target type for new() expression
+                //         (C _, C _) = new();
+                Diagnostic(ErrorCode.ERR_TypelessNewNoTargetType, "new()").WithLocation(8, 22),
+                // (8,22): error CS8131: Deconstruct assignment requires an expression with a type on the right-hand-side.
+                //         (C _, C _) = new();
+                Diagnostic(ErrorCode.ERR_DeconstructRequiresExpression, "new()").WithLocation(8, 22)
+                );
+        }
+
+        [Fact]
+        public void TestBestType_NullCoalescing()
+        {
+            var source = @"
+using System;
+
+class C
+{
+    public static void Main()
+    {
+        C c = null;
+        Console.Write(c ?? new());
```

Comment:
Please test with `new()` on the left side.

Reply:
`TestBinaryOperators02`